### PR TITLE
API-2008: Add new 'How to get PIM category tree' guided tutorial

### PR DESCRIPTION
**Pull request by LevFlavien**
https://akeneo.atlassian.net/browse/API-1907
https://akeneo.atlassian.net/browse/API-2008

https://github.com/akeneo/php-guided-tutorials/pull/8
https://github.com/akeneo/js-guided-tutorials/pull/7

### CODE REVIEW

File: content/tutorials/guides/how-to-get-pim-category-tree.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+Collect flat category tree from PIM API:
+
+```php [activate:PHP]
+
+function getCategories(): void
+{
+    $client = buildApiClient();
+
+    $rootCategoryCode = 'master';
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-get-pim-category-tree.md:121]: Get from the storage

File: content/tutorials/guides/how-to-get-pim-category-tree.md [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    } while (
+        $nextUrl
+    );
+
+		// Adds root category to complete the list.
+    $categoryPages[][] = getRootCategory($rootCategoryCode);
+
+    $categories = array_merge(...$categoryPages);
+
+    saveCategories($categories);
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-get-pim-category-tree.md:150]: add comments: the user doesn't know what is this function

File: content/tutorials/guides/how-to-get-pim-category-tree.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    }
+
+    return $rootCategory;
+}
+```
+
+```javascript [activate:NodeJS]
+
+async function fetchCategories() {
+  const rootCategoryCode = 'master';
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-get-pim-category-tree.md:173]: get from the sotrage

File: content/tutorials/guides/how-to-get-pim-category-tree.md [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    categories.push(...newCategories);
+
+    nextUrl = data._links?.next?.href;
+  } while (nextUrl)
+
+  // Adds root category to complete the list.
+  const rootCategory = await getRootCategory(rootCategoryCode);
+  categories.push(rootCategory);
+
+  saveCategories(categories);
```
Comment (tomglvng) [inline content/tutorials/guides/how-to-get-pim-category-tree.md:197]: add comments: the user doesn't know what is this function

**Review by tomglvng — CHANGES_REQUESTED** [review]
Fix comments

File: content/tutorials/guides/how-to-get-pim-category-tree.md (hunk 2 above)
Comment (LevFlavien) [reply content/tutorials/guides/how-to-get-pim-category-tree.md:150]: php-guided-tutorial: done

File: content/tutorials/guides/how-to-get-pim-category-tree.md (hunk 1 above)
Comment (LevFlavien) [reply content/tutorials/guides/how-to-get-pim-category-tree.md:121]: php-guided-tutorial: done

File: content/tutorials/guides/how-to-get-pim-category-tree.md (hunk 3 above)
Comment (LevFlavien) [reply content/tutorials/guides/how-to-get-pim-category-tree.md:176]: js-guided-tutorials: done

File: content/tutorials/guides/how-to-get-pim-category-tree.md (hunk 4 above)
Comment (LevFlavien) [reply content/tutorials/guides/how-to-get-pim-category-tree.md:200]: js-guided-tutorials: done

**Review by tomglvng — APPROVED** [review]

**Pull request merged**